README.md

```diff
@@ -48,69 +98,375 @@ These bindings allow Azure Functions to integrate with [Azure SignalR Service](h
+
+##### IServiceHubContextStore
+
+`IServiceHubContextStore` stores `IServiceHubContext` for each hub name.
+
+`ValueTask<IServiceHubContext> GetAsync(string hubName)`
+
+Gets `IServiceHubContext`. If the `IServiceHubContext` for a specific hub name exists, returns the `IServiceHubContext`, otherwise creates one and then returns it. `hubName` is the hub name of the `IServiceHubContext`. The returned value is an instance of `IServiceHubContext`.
+
+`IServiceManager ServiceManager`
+
+`IServiceManager` is used to create `IServiceHubContext`.
+
+For more introduction about `IServiceHubContext` and `IServiceManager`, please refer to [Azure SignalR Service Management SDK Guide](https://github.com/Azure/azure-signalr/blob/dev/docs/management-sdk-guide.md).
+
+> Note: We recommend you use `StaticServiceHubContextStore` instead of importing [SignalR Service Management package](https://www.nuget.org/packages/Microsoft.Azure.SignalR.Management). If you import [SignalR Service Management package](https://www.nuget.org/packages/Microsoft.Azure.SignalR.Management) directly, and you use `Persistent` [transport type](https://github.com/Azure/azure-signalr/blob/dev/docs/management-sdk-guide.md#transport-type), please don't create a `IServiceHubContext` and dispose it in a function method. Because for each http request, the function will be invoked. Therefore a new Websockets connection will be established and disconnected for each http request. That will have large impact to you function performance. Instead, you should make the `IServiceHubContext` a singleton to the function class. 
+
+The following are sample for negotiate function without SignalR input binding and broadcast messages without output binding respectively.
+
+##### Negotiate function without SignalR input binding
+
+With `StaticServiceHubContextStore`, you can generate `SignalRConnectionInfo` inside the function, provide user id, hub name and additional claims extracted from http request. 
+
+```C#
+[FunctionName("negotiate")]
+public static SignalRConnectionInfo GetSignalRInfo(
+    [HttpTrigger(AuthorizationLevel.Anonymous)] HttpRequest req)
+{
+    var userId = req.Query["userid"];
+    var hubName = req.Query["hubname"];
+    var connectionInfo = new SignalRConnectionInfo();
+    var serviceManager = StaticServiceHubContextStore.Get().ServiceManager;
+    connectionInfo.AccessToken = serviceManager
+        .GenerateClientAccessToken(
+            hubName,
+            userId,
+            new List<Claim> { new Claim("claimType", "claimValue") });
+    connectionInfo.Url = serviceManager.GetClientEndpoint(hubName);
+    return connectionInfo;
+}
+```
+
+##### Broadcast messages with output binding
+
+`IServiceHubContext` provide [Microsoft.Azure.SignalR](https://www.nuget.org/packages/Microsoft.Azure.SignalR/)-extended interfaces, if you are familiar how to use `IHubContext` to send messages in an app server, it will be easy for you to use `IServiceHubContext`.
```

Comment:
>-extended [](start = 111, length = 9)

is `-` desired?

Reply:
yes, I want to express that `IServiceHubContext` extends `IHubContext`